### PR TITLE
ARRAY_SIZE and unused variable

### PR DESCRIPTION
**Pull request by borneoa**
I have split the patch for ARRAY_SIZE in two parts:
the first covers only code not upstreamed and should be applied on riscv repository only
the second covers the code already upstreamed and should be applied on both riscv and upstream repositories.

### CODE REVIEW

**Comment by borneoa** [comment]
Two of these patches already pushed on upstream gerrit as:
http://openocd.zylin.com/6257
http://openocd.zylin.com/6258
but are both waiting for review/approval here

**Review by wxjstz — APPROVED** [review]

**Review by wxjstz — APPROVED** [review]

**Review by wxjstz — APPROVED** [review]

**Review by JanMatCodasip — APPROVED** [review]
Thank you, the changes are fine.

**Review by timsifive — APPROVED** [review]
Thank you for doing this cleanup.

**Pull request merged**
